cupy/linalg/einsum.py

```diff
@@ -315,6 +315,41 @@ def _get_out_shape(shape0, sub0, shape1, sub1, sub_out):
     return out_shape
 
 
+def _reshape_transpose(arr, mode, mode_out):
```

Comment:
I would name the function `_expand_dims_transpose` or `_unsqueeze_transpose` to clarify what kind of reshape is performed.

Reply:
Thanks. It looks `_exapnd_dims_transpose` is better. will change the name.